platforms/documentation/docs/src/docs/userguide/platforms/jvm/toolchains.adoc

```diff
@@ -128,6 +145,139 @@ A specification without a language version, in most cases, would be treated as a
 
 Usage of _invalid_ instances of `JavaToolchainSpec` results in a build error since Gradle 8.0.
 
+[[sec:source-target-toolchain]]
+=== 2. Source and Target compatibility
+
+Setting `sourceCompatibility` and `targetCompatibility` tells the Java compiler to produce bytecode compatible with a specific Java version but does *not* enforce which JDK Gradle itself runs with:
+
+[source,kotlin]
```

Comment:
🤔  Only the newly added snippets seem to not have the Kotlin + Groovy version. Should we make it consistent across the page and add Groovy?